python/ray/data/tests/test_dataset.py

```diff
@@ -1494,17 +1474,18 @@ def test_schema(ray_start_regular_shared):
 
 def test_schema_lazy(ray_start_regular_shared):
     ds = ray.data.range(100, parallelism=10)
-    # We kick off the read task for the first block by default.
-    assert ds._plan._in_blocks._num_computed() == 1
+    # We do not kick off the read task by default.
+    assert ds._plan._in_blocks._num_computed() == 0
     schema = ds.schema()
     assert schema == int
+    assert ds._plan._in_blocks._num_computed() == 1
     # Fetching the schema should not trigger execution of extra read tasks.
     assert ds._plan.execute()._num_computed() == 1
```

Comment:
cc @clarkzinzow FYI for question in https://github.com/ray-project/ray/pull/31558#discussion_r1066292322 .